tabular/tests/conftest.py

```diff
@@ -19,3 +27,115 @@ def pytest_collection_modifyitems(config, items):
     for item in items:
         if "slow" in item.keywords:
             item.add_marker(skip_slow)
+
+
+class DatasetLoaderHelper:
+    dataset_info_dict = dict(
+        # Binary dataset
+        adult={
+            'url': 'https://autogluon.s3.amazonaws.com/datasets/AdultIncomeBinaryClassification.zip',
+            'name': 'AdultIncomeBinaryClassification',
+            'problem_type': BINARY,
+            'label_column': 'class',
+        },
+        # Multiclass big dataset with 7 classes, all features are numeric. Runs SLOW.
+        covertype={
+            'url': 'https://autogluon.s3.amazonaws.com/datasets/CoverTypeMulticlassClassification.zip',
+            'name': 'CoverTypeMulticlassClassification',
+            'problem_type': MULTICLASS,
+            'label_column': 'Cover_Type',
+        },
+        # Regression with mixed feature-types, skewed Y-values.
+        ames={
+            'url': 'https://autogluon.s3.amazonaws.com/datasets/AmesHousingPriceRegression.zip',
+            'name': 'AmesHousingPriceRegression',
+            'problem_type': REGRESSION,
+            'label_column': 'SalePrice',
+        },
+    )
+
+    @staticmethod
+    def load_dataset(name: str, directory_prefix: str = './datasets/'):
+        dataset_info = copy.deepcopy(DatasetLoaderHelper.dataset_info_dict[name])
+        train_file = dataset_info.pop('train_file', 'train_data.csv')
+        test_file = dataset_info.pop('test_file', 'test_data.csv')
+        name_inner = dataset_info.pop('name')
+        url = dataset_info.pop('url', None)
+        train_data, test_data = DatasetLoaderHelper.load_data(
+            directory_prefix=directory_prefix,
+            train_file=train_file,
+            test_file=test_file,
+            name=name_inner,
+            url=url,
+        )
+
+        return train_data, test_data, dataset_info
+
+    @staticmethod
+    def load_data(directory_prefix, train_file, test_file, name, url=None):
+        if not os.path.exists(directory_prefix):
+            os.mkdir(directory_prefix)
+        directory = directory_prefix + name + "/"
+        train_file_path = directory + train_file
+        test_file_path = directory + test_file
+        if (not os.path.exists(train_file_path)) or (not os.path.exists(test_file_path)):
+            # fetch files from s3:
+            print("%s data not found locally, so fetching from %s" % (name, url))
+            zip_name = ag.download(url, directory_prefix)
+            ag.unzip(zip_name, directory_prefix)
+            os.remove(zip_name)
+
+        train_data = task.Dataset(file_path=train_file_path)
+        test_data = task.Dataset(file_path=test_file_path)
+        return train_data, test_data
+
+
+class FitHelper:
+    @staticmethod
+    def fit_and_validate_dataset(dataset_name, fit_args, sample_size=1000, refit_full=True, delete_directory=True):
```

Comment:
do we want time_limits for potentially slow models?

Reply:
I considered it, but I think we should only do this if unit tests become slow, since the alternative is the test could fail because the model didn't have enough time to train and didn't return a result.